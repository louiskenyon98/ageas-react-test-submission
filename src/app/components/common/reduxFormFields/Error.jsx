import React from 'react';
import PropTypes from 'prop-types';

class Error extends React.PureComponent {
    render() {
        if (this.props.message) {
            return (
                <div className="ui error message">
                    <div className="header">{this.props.message}</div>
                </div>
            )
        }
        return null
    }
}

Error.propTypes = {
    message: PropTypes.string
};

export default Error;
