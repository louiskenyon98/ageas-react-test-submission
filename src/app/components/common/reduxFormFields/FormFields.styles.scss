.formTitleField {
  border: none !important;
  font-size: 1.5em !important;
  text-align: center !important;
  background-color: #2980b9 !important;
  padding-left: 10px !important;
  padding-right: 10px !important;
  font-family: 'Thasadith', sans-serif !important;
  :focus {
    color: #fff !important;
  }
}
.formBodyFieldWrapper {
  .formBodyField {
    border: none !important;
    font-size: 1.5em !important;
    font-family: 'Thasadith', sans-serif !important;
    resize: none;
  }

}
