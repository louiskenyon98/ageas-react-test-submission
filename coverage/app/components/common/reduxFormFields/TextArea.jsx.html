<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/common/reduxFormFields/TextArea.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">app/components/common/reduxFormFields</a> TextArea.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import Error from './Error';
import styles from '../../../../themes/style.scss';
&nbsp;
export class TextArea extends React.Component {
&nbsp;
    hasError() {
        const {touched, error} = this.props.meta;
        return (touched &amp;&amp; error);
    }
&nbsp;
    render() {
        const className = `field ${this.hasError() ? 'error' : <span class="branch-1 cbranch-no" title="branch not covered" >''}</span>`;
        return (
            &lt;div className={`${className} ${styles["form-body-field-wrapper"]}`}&gt;
                &lt;textarea {...this.props.input} placeholder={this.props.label} rows="10" cols="40" className={`${styles["form-body-field"]}`} /&gt;
                {this.hasError() &amp;&amp;
                &lt;Error
                    message={this.props.meta.error}
                /&gt;}
            &lt;/div&gt;
        )
    }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 17 2019 15:42:32 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
