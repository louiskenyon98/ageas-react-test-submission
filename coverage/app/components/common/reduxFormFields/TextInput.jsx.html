<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/common/reduxFormFields/TextInput.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">app/components/common/reduxFormFields</a> TextInput.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/4</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import Error from './Error';
import styles from "../../../../themes/style.scss";
&nbsp;
export class TextInput extends React.Component {
&nbsp;
<span class="fstat-no" title="function not covered" >    ha</span>sError() {
        const {touched, error} = <span class="cstat-no" title="statement not covered" >this.props.meta;</span>
<span class="cstat-no" title="statement not covered" >        return (touched &amp;&amp; error);</span>
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    re</span>nder() {
        const className = <span class="cstat-no" title="statement not covered" >`field ${this.hasError() ? 'error' : ''}`;</span>
<span class="cstat-no" title="statement not covered" >        return (</span>
            &lt;div className={`${className} ${styles["form-body-field-wrapper"]}`}&gt;
              &lt;input {...this.props.input} placeholder={this.props.label} autoComplete="off" className={`${styles["form-title-field"]}`} /&gt;
                {this.hasError() &amp;&amp;
                &lt;Error
                    message={this.props.meta.error}
                /&gt;}
            &lt;/div&gt;
        )
    }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 24 2019 11:41:42 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
