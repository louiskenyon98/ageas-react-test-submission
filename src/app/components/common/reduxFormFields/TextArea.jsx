import React from 'react';
import Error from './Error';
import styles from "./FormFields.styles.scss";

export class TextArea extends React.Component {

    hasError() {
        const {touched, error} = this.props.meta;
        return (touched && error);
    }

    render() {
        const className = `field ${this.hasError() ? 'error' : ''}`;
        return (
            <div className={`${className} ${styles.formBodyFieldWrapper}`}>
                <textarea {...this.props.input} placeholder={this.props.label} rows="10" cols="40" className={`${styles.formBodyField}`} />
                {this.hasError() &&
                <Error
                    message={this.props.meta.error}
                />}
            </div>
        )
    }
}
