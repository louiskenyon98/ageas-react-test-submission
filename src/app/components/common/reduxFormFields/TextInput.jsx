import React from 'react';
import Error from './Error';
import styles from "./FormFields.styles.scss";

export class TextInput extends React.Component {

    hasError() {
        const {touched, error} = this.props.meta;
        return (touched && error);
    }

    render() {
        const className = `field ${this.hasError() ? 'error' : ''}`;
        return (
            <div className={`${className} ${styles.formBodyFieldWrapper}`}>
              <input {...this.props.input} placeholder={this.props.label} autoComplete="off" className={`${styles.formTitleField}`} />
                {this.hasError() &&
                <Error
                    message={this.props.meta.error}
                />}
            </div>
        )
    }
}
