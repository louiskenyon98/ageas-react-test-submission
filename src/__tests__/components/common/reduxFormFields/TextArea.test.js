import React from 'react';
import {shallow} from 'enzyme';
import {TextArea} from '../../../../app/components/common/reduxFormFields/TextArea';

describe('TextArea', () => {
    let props = {};
    let wrapper = () => shallow(<TextArea {...props}/>);
    beforeEach(() => {
        props = {
            meta: {
                touched: true,
                error: 'Error string'
            },
            label: 'Placeholder string',
            input: {
                foo: 'bar'
            }
        }
    });
    describe('render', () => {
        it('should render textarea with placeholder and text in error state if touched and error = true', () => {
            expect(wrapper()).toMatchSnapshot();
        })
    })
});
