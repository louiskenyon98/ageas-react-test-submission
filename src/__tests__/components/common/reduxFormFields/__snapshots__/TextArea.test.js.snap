// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea render should render textarea in error state if touched and error = true 1`] = `
<div
  className="field error form-body-field-wrapper"
>
  <textarea
    className="form-body-field"
    cols="40"
    foo="bar"
    placeholder="Placeholder string"
    rows="10"
  />
  <Error
    message="Error string"
  />
</div>
`;

exports[`TextArea render should render textarea with placeholder and text in error state if touched and error = true 1`] = `
<div
  className="field error form-body-field-wrapper"
>
  <textarea
    className="form-body-field"
    cols="40"
    foo="bar"
    placeholder="Placeholder string"
    rows="10"
  />
  <Error
    message="Error string"
  />
</div>
`;
